### PR TITLE
feat: Exclude experimental guidance from standard channel CRDs

### LINKED ISSUE #2160: CRD Generation: Exclude Experimental Guidance from Standard Channel CRDs
**What would you like to be added**:
Today we can choose to specify different validation and more fields for Experimental Channel CRDs. Unfortunately we don't have a way to provide different descriptions depending on the channel. 

**Why this is needed**:
This came up in https://github.com/kubernetes-sigs/gateway-api/pull/2157 when the overall information about Gateway Addresses mentions the need to set Routability, despite that field only being present in the Experimental channel.

**Notes:**
For anyone looking to work on this, we'd need to modify/extend something around here and add some kind of start and stop tags for content that was only intended for the experimental channel:

https://github.com/kubernetes-sigs/gateway-api/blob/f883de997b88dd6ee138930198542da8a9b2f634/pkg/generator/main.go#L149C1-L170

### PR DESCRIPTION
**Pull request by slayer321**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
This PR adds the feature where we can specifiy the start and end tag to the CRDs description which only need to be there in expremental channel CRDs and exclude from standard channel CRDs.

**Which issue(s) this PR fixes**:

Fixes #2160

**Does this PR introduce a user-facing change?**:

```release-note
Excluding experimental guidance in field descriptions from standard CRDs
```

### CODE REVIEW

**Comment by slayer321** [comment]
I raised this PR which already have Routability changes, and after raising as there was a merge conflict I  found that the Routability changes where drop. So the changes I made I tested with Routability part and can confirm it works.

**Review by robscott — COMMENTED** [review]
Thanks @slayer321! This looks good to me, but it would be helpful if you could start by wrapping some examples so we can confirm it works as intended. Here are the main ones I'm thinking of:

* https://github.com/kubernetes-sigs/gateway-api/blob/73d727a9a4d69a4a22b7cf073d962ad820e2ccfb/apis/v1beta1/shared_types.go#L28 
* https://github.com/kubernetes-sigs/gateway-api/blob/73d727a9a4d69a4a22b7cf073d962ad820e2ccfb/apis/v1beta1/shared_types.go#L69-L77
* https://github.com/kubernetes-sigs/gateway-api/blob/73d727a9a4d69a4a22b7cf073d962ad820e2ccfb/apis/v1beta1/shared_types.go#L130-L132
* https://github.com/kubernetes-sigs/gateway-api/blob/73d727a9a4d69a4a22b7cf073d962ad820e2ccfb/apis/v1beta1/shared_types.go#L170
* https://github.com/kubernetes-sigs/gateway-api/blob/73d727a9a4d69a4a22b7cf073d962ad820e2ccfb/apis/v1beta1/shared_types.go#L192-L200

File: apis/v1beta1/shared_types.go [hunk 1]
```diff
@@ -66,6 +66,7 @@ type ParentReference struct {
 	// Gateway has the AllowedRoutes field, and ReferenceGrant provides a
 	// generic way to enable any other kind of cross-namespace reference.
 	//
+	//<experimental:description>
```
Comment (robscott) [inline apis/v1beta1/shared_types.go:69]: One tiny nit, probably best to stick with the `<gateway:experimental` prefix we're using elsewhere.
```suggestion
	// <gateway:experimental:description>
```

**Review by robscott — APPROVED** [review]
Thanks @slayer321! Sorry I lost track of this one!

**Comment by slayer321** [comment]
@robscott I have made the changes can you please take a look.

**Comment by robscott** [comment]
Thanks @slayer321, Sorry I lost track of this one! Please add a release-note in the PR description above, maybe something like "Excluding experimental guidance in field descriptions from standard CRDs". Once you do that, I believe the PR should merge.

/lgtm
/approve

**Pull request merged**